File: C++/1793. Maximum Score of a Good Subarray.cpp
class Solution {
public:
    int maximumScore(vector<int>& nums, int k) {
        int a, b, i = k, j = k, n = nums.size(), res = nums[k], t = nums[k], m = 2;

        while (i > 0 || j < n - 1) {
            a = (i == 0) ? 0 : nums[i - 1];
            b = (j == n - 1) ? 0 : nums[j + 1];

            a < b ? j++ : i--;

            t = min(t, min(nums[i], nums[j]));
            res = max(res, t * m++);
        }

        return res;
    }
};

File: Python/1793. Maximum Score of a Good Subarray.py
class Solution:
    def maximumScore(self, nums: List[int], k: int) -> int:
        res = t = nums[k]
        i, j, n, m = k, k, len(nums), 2

        while i > 0 or j < n - 1:
            if (nums[i - 1] if i > 0 else 0) < (nums[j + 1] if j < n - 1 else 0): j += 1
            else: i -= 1

            t = min(t, nums[i], nums[j])
            res = max(res, t * m)
            m += 1

        return res
